src/main/scala/org/broadinstitute/dsde/agora/server/webservice/routes/OpenIdConnectDirectives.scala

```diff
@@ -2,25 +2,36 @@ package org.broadinstitute.dsde.agora.server.webservice.routes
 
 import akka.http.scaladsl.server.Directive1
 import akka.http.scaladsl.server.Directives._
+import org.broadinstitute.dsde.agora.server.dataaccess.SamClient
 import org.broadinstitute.dsde.agora.server.webservice.util.ImplicitMagnet
 
 import scala.concurrent.ExecutionContext
 
 
 /**
  * Directives class to retrieve user information from the OpenId Connect proxy
- * provided information in the header
+ * provided information in the header.
  */
 class OpenIdConnectDirectives extends AgoraDirectives {
+  private val serviceAccountDomain = "\\S+@\\S+\\.iam\\.gserviceaccount\\.com".r
+
+  private def isServiceAccount(email: String) = {
+    serviceAccountDomain.pattern.matcher(email).matches
+  }
 
   override def commonNameFromRequest(magnet: ImplicitMagnet[ExecutionContext]): Directive1[String] = {
     headerValueByName("OIDC_CLAIM_name")
```

Comment:
In the case of the switcharoo, what should the name be?

Reply:
I don't know but this is unused anyway so I will delete